### ISSUE TITLE
ImportOrder: make imports ordering as in Eclipse

### ISSUE THREAD

**Issue by HSerg** [checkstyle/checkstyle#4338]
```
/var/tmp $ javac -cp lombok.jar Test.java
```

```
/var/tmp $ cat Test.java
import lombok.val;
import lombok.experimental.FieldDefaults;

@FieldDefaults(makeFinal = true)
class Test {
        void main() {
                val value = 5;
        }
}
```
```
/var/tmp $ cat checkstyle.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="ImportOrder">
            <property name="groups" value="/^java\./,javax,org,com"/>
            <property name="ordered" value="true"/>
            <property name="separated" value="true"/>
            <property name="option" value="top"/>
            <property name="caseSensitive" value="true"/>
            <property name="sortStaticImportsAlphabetically" value="true"/>
            <property name="useContainerOrderingForStatic" value="true"/>
        </module>
    </module>
</module>
```
```
/var/tmp $ java -jar checkstyle-7.7-all.jar -c checkstyle.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:2: Wrong order for 'lombok.experimental.FieldDefaults' import. [ImportOrder]
Audit done.
Checkstyle ends with 1 errors.
```

Expected: no violations for Eclipse users.

Relates to https://github.com/checkstyle/checkstyle/issues/3101

**Comment by romani** [comment]
@HSerg , what version (exact version is required) of eclipse do you use ?

**Comment by HSerg** [comment]
Neon.3 Release (4.6.3)

**Comment by romani** [comment]
@HSerg , why you think Eclipse is right in his sorting ?
"lombok.e..." should be before "lombok.v..."

**Comment by HSerg** [comment]
Eclipse style sorting order - compare package names, then class names. See [ImportOrderCheck.compareContainerOrder()](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java#L565)
"lombok" should be before "lombok.experimental".

**Comment by romani** [comment]
Issue is approved, but investigation need to be done on "Container ordering".... is it only Eclipse mode of sorting? I am not remember why we did this ordering only for statics few years ago.....

We use that type of sorting for statics now only, probably we need new property to do this for other imports. Special attention should be done on documentation of this new feature.

**Comment by MEZk** [comment]
@romani 
>  I am not remember why we did this ordering only for statics few years ago.....

There was [the reported bug](https://bugs.eclipse.org/bugs/show_bug.cgi?id=473629) in Eclipse which  referred to static imports only. We decided to implement that feature only for static imports as was requested and did not touch unrelated check's logic.

**Comment by romani** [comment]
One more point why we missed it, and never run into it before. 
We never worked with sources that have classes named from lower character `lombok.val` :) .

**Comment by ricbit** [comment]
I'm hitting this bug on my project. If no one is working on this bug, I offer to fix it.

**Comment by romani** [comment]
@ricbit , I do not remember any activity on this issue, please be welcome with Pull Request.
